docs/concepts/decorators-and-auto-logging.md

```diff
@@ -0,0 +1,106 @@
+import pandas as pd
+from sklearn.svm import SVC
+import mlrun
+
+@mlrun.handler(labels={'framework':'scikit-learn'},
+               outputs=['prediction:dataset'],
+               inputs={"train_data": pd.DataFrame,
+                       "predict_input": pd.DataFrame})
+def train_and_predict(train_data,
+                      predict_input,
+                      label_column='label'):
+
+    x = train_data.drop(label_column, axis=1)
+    y = train_data[label_column]
+
+    clf = SVC()
+    clf.fit(x, y)
+
+    return list(clf.predict(predict_input))
+```
+
+To run the code, use the following example:
+
+``` python
+import mlrun
+project = mlrun.get_or_create_project("mlrun-example", context="./", user_project=True)
+
+trainer = project.set_function("train.py", name="train_and_predict", kind="job", image="mlrun/mlrun", handler="train_and_predict")
+
+trainer_run = project.run_function(
+    "train_and_predict", 
+    inputs={"train_data": mlrun.get_sample_path('data/iris/iris_dataset.csv'),
+            "predict_input": mlrun.get_sample_path('data/iris/iris_to_predict.csv')
+           }
+)
+```
+
+The outcome will be a run with:
+1. A label with key "framework" and value "scikit-learn".
+2. Two inputs "train_data" and "predict_input" created from Pandas DataFrame.
+3. An artifact called "prediction" of type "dataset". The contents of the dataset will be the return value (in this case the prediction result).
+
+## Labels
+
+The decorator gives you the option to set labels for the run. The `labels` parameter is a dictionary with keys and values to set for the labels.
+
+## Input type parsing
+
+The `mlrun.handler` decorator can also parse the input types, if they are specified, an equivalent definition is as follows:
+
+``` python
+@mlrun.handler(labels={'framework':'scikit-learn'},
+               outputs=['prediction:dataset'])
+def train_and_predict(train_data: pd.DataFrame,
+                      predict_input: pd.DataFrame,
+                      label_column='label'):
+
+...
+```
+
+> **Note:** If the inputs does not have a type input, the decorator would assume the parameter type in {py:class}`mlrun.datastore.DataItem`. Similarly, if you specify `inputs=False`, all the run inputs are assumed to be of type `mlrun.datastore.DataItem`.
+
+## Logging return values as artifacts
+
+If you specify the `outputs` parameter, the return values will be logged as the run artifacts. `outputs` expects a list, the length of the list must match the number of returned values.
+
+The simplest option is to specify a list of strings. Each string containers the name of the artifact. You can also specify the artifact type by adding a colon after the artifact name followed by the type (`'name:artifact_type'`). The following are valid artifact types:
+
+- dataset
+- directory
+- file
+- object
+- plot
+- result
+
+Another option is to specify a tuple in the form of `(name, artifact_type, arguments)`. Refer to the {py:func}`mlrun.handler` for more details.
```

Comment:
`tuple` can be `(name, artifact_type)` as well (`arguments` is not mandatory). Also, the simple `name` can be given as well and there is a default type to artifact type dictionary that is mapping the object type to be logged as its default artifact. For example, if the returned object is a `pd.DataFrame` it will be logged automatically as a `DatasetArtifact`.

```
DEFAULT_OBJECTS_ARTIFACT_TYPES_MAP = {
    pd.DataFrame: ArtifactType.DATASET,
    pd.Series: ArtifactType.DATASET,
    np.ndarray: ArtifactType.DATASET,
    dict: ArtifactType.RESULT,
    list: ArtifactType.RESULT,
    tuple: ArtifactType.RESULT,
    str: ArtifactType.RESULT,
    int: ArtifactType.RESULT,
    float: ArtifactType.RESULT,
    bytes: ArtifactType.OBJECT,
    bytearray: ArtifactType.OBJECT,
    plt.Figure: ArtifactType.PLOT,
    plotly.graph_objs.Figure: ArtifactType.PLOT,
    bokeh.plotting.Figure: ArtifactType.PLOT,
}
```